src/compositions/useFlowRuns.ts

```diff
@@ -26,7 +32,16 @@ export function useFlowRuns(filter: FlowRunsFilter | Ref<FlowRunsFilter | null |
       return null
     }
 
-    return [filterRef.value]
+    const limit = filterRef.value.limit ?? FLOW_RUN_LIMIT
+    const offset = (pageRef.value - 1) * limit
+
+    const filter: FlowRunsFilter = {
+      ...filterRef.value,
+      limit,
+      offset,
+    }
```

Comment:
Now that I see this on (paper) I'm realizing this could probably be even further abstracted since it's not specific to the flow runs route - maybe something like:
```ts
const GLOBAL_API_LIMIT = 200

function useFilterPagination(page: MaybeRef<number> = 1, limit: MaybeRef<number> = GLOBAL_API_LIMIT) {
    const pageRef = ref(page)
    const limitRef = ref(limit)
    const offset = computed(() => (pageRef.value - 1) * limit.value)

    return {
      limit,
      offset,
    }
}

export function useFlowRuns(filter: FlowRunsFilter | Ref<FlowRunsFilter | null | undefined>, page: MaybeRef<number> = 1, options?: SubscriptionOptions): UseFlowRuns {
  const { limit, offset } = useFilterPagination(page, filterRef.value.limit)
}

```

It's a small change to how you're juggling the filter (you might need to create a ref for it?) but it'll let you re-use the logic elsewhere, since the API limits are only applied globally anyway

Reply:
Great suggestion. Updated to use this pattern.